core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala

```diff
@@ -53,6 +54,9 @@ private[ui] class ExecutorsPage(
   // When GCTimePercent is edited change ToolTips.TASK_TIME to match
   private val GCTimePercent = 0.1
 
+  // a safe String to Int for sorting ids (converts non-numeric Strings to -1)
+  private def strToInt(str: String) : Int = Try(str.toInt).getOrElse(-1)
```

Comment:
This makes all non-numeric strings "equal" and their sort order arbitrary. I think "driver" is probably the only non-numeric value in use and that's unlikely to change, so this is probably OK.